builtin/emerge/env.lua

```diff
@@ -47,12 +47,48 @@ function core.get_perlin(seed, octaves, persist, spread)
 		}
 	end
 	params.seed = core.get_seed(params.seed) -- add mapgen seed
-	return PerlinNoise(params)
+	return ValueNoise(params)
 end
 
-
-function core.get_perlin_map(params, size)
+function core.get_value_noise_map(params, size)
 	local params2 = table.copy(params)
 	params2.seed = core.get_seed(params.seed) -- add mapgen seed
-	return PerlinNoiseMap(params2, size)
+	return ValueNoiseMap(params2, size)
+end
+
+-- deprecated as of 5.12, as it was not Perlin noise
+local get_perlin_deprecation_message_printed = false
+function core.get_perlin(seed, octaves, persist, spread)
+	if not get_perlin_deprecation_message_printed then
+		core.log("deprecated", "core.get_perlin is deprecated and was renamed to core.get_value_noise")
+		get_perlin_deprecation_message_printed = true
+	end
+	return core.get_value_noise(seed, octaves, persist, spread)
+end
+local get_perlin_map_deprecation_message_printed
+-- deprecated as of 5.12, as it was not Perlin noise
+function core.get_perlin_map(params, size)
+	if not get_perlin_map_deprecation_message_printed then
+		core.log("deprecated", "core.get_perlin_map is deprecated and was renamed to core.get_value_noise_map")
+		get_perlin_map_deprecation_message_printed = true
+	end
+	return core.get_value_noise_map(params, size)
+end
+-- deprecated as of 5.12, as it was not Perlin noise
+local perlinnoise_deprecation_message_printed = false
+function PerlinNoise(noiseparams_or_seed, octaves, persistence, spread)
+	if not perlinnoise_deprecation_message_printed then
+		core.log("deprecated", "PerlinNoise is deprecated and was renamed to ValueNoise")
+		perlinnoise_deprecation_message_printed = true
+	end
+	return ValueNoise(noiseparams_or_seed, octaves, persistence, spread)
+end
+-- deprecated as of 5.12, as it was not Perlin noise
+local perlinnoisemap_deprecation_message_printed = false
+function PerlinNoiseMap(noiseparams, size)
+	if not perlinnoisemap_deprecation_message_printed then
+		core.log("deprecated", "PerlinNoise is deprecated and was renamed to ValueNoise")
+		perlinnoisemap_deprecation_message_printed = true
+	end
+	return ValueNoiseMap(noiseparams, size)
 end
```

Comment:
I'm not convinced this deprecation warning brings any value.
Unless I'm writing a mod for 5.12 or later I now have to do the following:
```lua
local PerlinNoiseMap_ = rawget(_G, "ValueNoiseMap") or PerlinNoiseMap
local stuff = PerlinNoiseMap_(foo, bar)
```

this brings no value as we can't realistically reuse that name anyway. what's the way forward?